### PR TITLE
(docs): add docs for visual matchers

### PR DESCRIPTION
**Pull request by christian-bromann**
Adding documentation for visual matchers.

See implementation in https://github.com/webdriverio/visual-testing/pull/141

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -694,6 +694,63 @@ await expect(browser).toHaveUrl(/webdriver\.io/)
...
+
+##### Usage
+
+```js
+await expect(browser).toMatchScreenSnapshot('partialPage', 0, {
+    // options
+})
+```
+
+### toMatchScreenSnapshot
```
Comment (wswebcreation) [inline docs/API.md:730]:
```suggestion
### toMatchFullPageSnapshot
```

File: docs/API.md [hunk 2]
```diff
@@ -694,6 +694,63 @@ await expect(browser).toHaveUrl(/webdriver\.io/)
...
+
+##### Usage
+
+```js
+await expect(browser).toMatchFullPageSnapshot('fullPage', 0, {
+    // options
+})
+```
+
+### toMatchScreenSnapshot
```
Comment (wswebcreation) [inline docs/API.md:742]:
```suggestion
### toMatchTabbablePageSnapshot
```

**Review by wswebcreation — CHANGES_REQUESTED** [review]
Thanks, can you also refer to them in the Visual Docs?

**Comment by christian-bromann** [comment]
Enhanced visual docs with custom visual matcher:
Screenshot 2024-01-10 at 7 44 29 PM

@wswebcreation is this good to go?

**Review by wswebcreation — APPROVED** [review]
LGTM, thanks

**Pull request merged**